signac/synced_collections/validators.py

```diff
@@ -101,12 +133,8 @@ def json_format_validator(data):
         return
     elif switch_type == "MAPPING":
         for key, value in data.items():
-            # Support for non-str keys will be removed in version 2.0.
-            # See issue: https://github.com/glotzerlab/signac/issues/316.
-            if not isinstance(key, (str, int, bool, type(None))):
-                raise KeyTypeError(
-                    f"Keys must be str, int, bool or None, not {type(key).__name__}"
-                )
+            if not isinstance(key, str):
+                raise KeyTypeError(f"Keys must be str, not {type(key).__name__}")
```

Comment:
No, because in the relevant cases where we need to support this the `_convert_key_to_str` is now handling the conversion upstream. That is what the ordering comment regarding validators in `collection_json` is referring to.